### PR TITLE
Display "action step" blocks (like voter reg.) in affirmation modal.

### PR DESCRIPTION
**Pull request by DFurnes**
### What does this PR do?
This PR allows editors to place "action step" blocks into the affirmation modal. This allows us to push other actions than sharing the campaign, like registering to vote.

screen shot 2018-02-20 at 3 39 14 pm

And here's the same page [on tablet](https://user-images.githubusercontent.com/583202/36450824-36ef18f0-165d-11e8-9d1d-573d914a9ccb.png) & [on mobile](https://user-images.githubusercontent.com/583202/36450795-11a279c0-165d-11e8-8e65-43fbe152aa0c.png)!

### Any background context you want to provide?
This involved more refactoring than I originally estimated. In particular:

🌵 Extract a single generalized `ActionStepBlock` component out of the `ActionStepWrapper` (which is used to render the entire action page). 2172965

📳 Add `Affirmation` as a possible "action step" block, and then use the `ActionStepBlock` in the affirmation modal! This should keep existing behavior for current campaigns, but allows us to start to introduce all the fun new stuff here over time. 3ccc211

🏭 Pull "page-specific" elements (like built-in section headers & column layout wrappers) into the `ActionStepWrapper` component, so `ActionStepBlock`is _just_ the step itself. b76858e

💱 Creates a `ModalControls` component that overlays the "close card" button, so that we don't have to muddy the rest of our components by passing this down through the hierarchy. e7a0390

🆔 Finally, this makes _all_ steps on the action page linkable! (Not just section headers.) 967bb8e

### What are the relevant tickets/cards?
[#154732073](https://www.pivotaltracker.com/story/show/154732073)

### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.
- [x] Added screenshot to PR description of related front-end updates on **small** screens.
- [x] Added screenshot to PR description of related front-end updates on **medium** screens.
- [x] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

**Comment by DFurnes** [comment]
One nice side effect of pulling layout into the "action page" itself was some surprise design fixes:

layout-fix

**Comment by DFurnes** [comment]
I've added this review app to the Thor client, so feel free to test the signup flow [with the voter registration modal](https://ds-phoenix-staging-pr-693.herokuapp.com/us/campaigns/test-voters-for-jeans/community) or [with the standard affirmation block](https://ds-phoenix-staging-pr-693.herokuapp.com/us/campaigns/test-teens-for-jeans) on the review app.

**Comment by DFurnes** [comment]
@DoSomething/team-rocket Okay, this should be ready for review!

File: resources/assets/components/ActionPage/ActionStep.js [hunk 1]
```diff
@@ -29,32 +29,29 @@ const ActionStep = (props) => {
...
-            <FlexCell width="two-thirds">
-              <Markdown>{ content }</Markdown>
-            </FlexCell>
-            :
-            null
-          }
-          {photoComponent}
-        </Flex>
-      </div>
+    <div className={classnames('action-step', { '-truncate': shouldTruncate })}>
```
Comment (weerd) [inline resources/assets/components/ActionPage/ActionStep.js]: 🙌 removing some nested flex cells!

**Comment by weerd** [comment]
> This PR allows editors to place "action step" blocks into the affirmation modal. This allows us to push other actions than sharing the campaign, like registering to vote.

So this is actual `ActionStep` components... did we want specifically `Action` components since we want to move towards the idea of content blocks and `Action` that are self-contained? From the screenshots it looks more specifically like it's just the `Action` and not a full `ActionStep` but maybe this is not feasible in current structure, since we're in an "inbetween" spot... 🤔

File: resources/assets/components/ActionPage/ActionStep.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ const renderPhoto = (photo, index) => (
 
 const ActionStep = (props) => {
   const {
-    id, title, stepIndex, content, photos, photoWidth,
+    title, stepIndex, content, photos, photoWidth,
```
Comment (weerd) [inline resources/assets/components/ActionPage/ActionStep.js]: So we no longer want the `id`?

File: resources/assets/components/ActionPage/ActionStepsWrapper.js [hunk 3]
```diff
@@ -18,34 +53,40 @@ const ActionStepsWrapper = (props) => {
...
-        return renderShareAction(step);
+    // Is this a "numbered" step? If so, increment our step index.
+    if (['third-party-action', 'campaignActionStep', 'default'].includes(type)) {
+      stepIndex += 1;
+    }
 
-      case 'linkAction':
-        return renderLinkAction(step);
+    // Some components have built-in section headers. For those, append it.
+    // @TODO: These should be split out into separate "content" blocks.
```
Comment (weerd) [inline resources/assets/components/ActionPage/ActionStepsWrapper.js]: 👏 ok, I hadn't gotten to this point in the changes, but this addresses my earlier comment.

File: resources/assets/components/ActionPage/ActionStep.js (hunk 2 above)
Comment (DFurnes) [reply resources/assets/components/ActionPage/ActionStep.js]: It's no longer being used in this component, so I removed from the props.

**Comment by DFurnes** [comment]
> So this is actual `ActionStep` components... did we want specifically `Action` components since we want to move towards the idea of content blocks and `Action` that are self-contained? From the screenshots it looks more specifically like it's just the `Action` and not a full `ActionStep` but maybe this is not feasible in current structure, since we're in an "inbetween" spot... 🤔 

Yeah definitely a work-in-progress. Longer term, I'd love all these to render with the exact same `<Block>` component that we set up for the feed but I had to cut off my refactoring somewhere!! 😅

**Pull request merged**